packages/query/src/bitcoin/ordinals/inscriptions.query.ts

```diff
@@ -208,33 +212,38 @@ export function useGetInscriptionsInfiniteQuery({
   return query;
 }
 
+const bestinslotInscriptionsRequestNum = 2000;
+
 export function useInscriptionsByAddressQuery(address: string) {
   const network = useLeatherNetwork();
-  const limiter = useHiroApiRateLimiter();
+  const client = useBitcoinClient();
 
   const query = useInfiniteQuery({
     enabled: !!address,
     queryKey: [QueryPrefixes.InscriptionsByAddress, network.id, address],
     async queryFn({ pageParam, signal }) {
-      return limiter.add(
-        () =>
-          fetchInscriptions({
-            addresses: address,
-            offset: pageParam,
-            signal,
-          }),
-        { priority: 1, signal, throwOnTimeout: true }
-      );
+      const res = await client.BestinSlotApi.getInscriptionsByAddress({
+        address,
+        network: network.id as WalletDefaultNetworkConfigurationIds,
+        offset: pageParam,
+        signal,
+        count: bestinslotInscriptionsRequestNum,
+      });
+
+      return {
+        offset: pageParam,
+        data: res.data,
+      };
     },
     initialPageParam: 0,
     getNextPageParam(lastPage) {
-      if (lastPage.offset >= lastPage.total) return undefined;
-      return lastPage.offset + 60;
+      if (lastPage.data.length < bestinslotInscriptionsRequestNum) return undefined;
+      return lastPage.offset + bestinslotInscriptionsRequestNum;
     },
-    refetchOnMount: false,
-    refetchOnReconnect: false,
-    refetchOnWindowFocus: false,
-    staleTime: 3 * 60 * 1000,
+    // refetchOnMount: false,
+    // refetchOnReconnect: false,
+    // refetchOnWindowFocus: false,
+    // staleTime: 3 * 60 * 1000,
```

Comment:
Did you mean to leave these commented out?

Reply:
yes, I've reverted it in main pr